Continue slideshow after deleting the current image

--- a/src/slideshow.c
+++ b/src/slideshow.c
@@ -165,7 +165,7 @@ gboolean slideshow_should_continue(SlideShowData *ss)
 		imd_fd = image_get_fd(ss->imd);
 
 	if ( ((imd_fd == NULL) != (ss->slide_fd == NULL)) ||
-	    (imd_fd && ss->slide_fd && imd_fd != ss->slide_fd) ) return FALSE;
+	    (imd_fd && ss->slide_fd && imd_fd != ss->slide_fd) ) return TRUE;
 
 	if (ss->filelist) return TRUE;
 
